src/backend/cdb/cdbcat.c

```diff
@@ -458,8 +458,16 @@ GpPolicyFetch(Oid tbloid)
 				}
 
 				/* Create a GpPolicy object. */
-				policy = makeGpPolicy(POLICYTYPE_PARTITIONED,
-									  nattrs, policyform->numsegments);
+				if (policyform->numsegments == 0)
```

Comment:
why change this ？

Reply:
We support to create table on single master without any segments. We create segments through creating warehouse on single master, switch to warehouse and query.